### PR TITLE
Upstream sync 2024 06 16

### PR DESCRIPTION
**Pull request by robertgshaw2-neuralmagic**
Upstream sync 2024 06 16 (https://github.com/neuralmagic/nm-vllm/pull/310) - v0.5.0.post of vllm

SUMMARY:

* Merge commits from https://github.com/vllm-project/vllm/commit/8f89d72090da70895d77d32248ea8504f7daba50 to https://github.com/vllm-project/vllm/commit/0f0d8bc065f3608e7657a9696f5d2d7c0d6722d1
* Limit numpy to < 2.0
* Updated `run-tests` to print name of the test that is about to run (for debugging what hangs in automation)
* Disable usage stats in automation
* Temporarily disable ENTRYPOINTS (to be re-enabled in Andy's single whl PR)
* Updated `run-tests` to consider exit code 5 from pytest to be a pass (since exit code 5 from pytest means that we did not run any tests)

Note that https://github.com/vllm-project/vllm/commit/8f89d72090da70895d77d32248ea8504f7daba50 is NOT included in this merge.

COMPARE vs UPSTREAM:

https://github.com/neuralmagic/nm-vllm/compare/upstream-sync-2024-06-16..vllm-project:vllm:v0.5.0.post1

### CODE REVIEW

File: .github/scripts/run-tests [hunk 1]
```diff
@@ -109,6 +109,10 @@ do
     LOCAL_SUCCESS=0
     RESULT_XML=$(echo ${TEST} | sed -e "s/${TEST_DIR}/${RESULTS_DIR}/" | sed -e "s/.py/.xml/")
 
+    # report which test is being run
+    # (in CI, if a test hangs, this logs *which* test is running *before* it hangs)
+    echo "RUNNING TEST: ${TEST}"
```
Comment (andy-neuma) [inline .github/scripts/run-tests:114]: nice

hopefully, we can get rid of this script soon.

File: vllm/__init__.py [hunk 2]
```diff
@@ -12,10 +12,10 @@
 from vllm.pooling_params import PoolingParams
 from vllm.sampling_params import SamplingParams
 
-# UPSTREAM SYNC: use the current downstream.
-__version__ = "0.5.0"
+from .version import __version__
 
 __all__ = [
+    "__version__",
```
Comment (andy-neuma) [inline vllm/__init__.py:18]: where did the version's value go?

**Review by andy-neuma — APPROVED** [review]
thanks

File: vllm/__init__.py (hunk 2 above)
Comment (robertgshaw2-neuralmagic) [reply vllm/__init__.py:18]: Its moved to `version.py` + then imported

File: .github/scripts/run-tests [hunk 3]
```diff
@@ -125,7 +129,20 @@ do
         pytest ${CC_PYTEST_FLAGS} --junitxml=${RESULT_XML} ${TEST} || LOCAL_SUCCESS=$?
     fi
 
-    SUCCESS=$((SUCCESS + LOCAL_SUCCESS))
+    # if a file gets exit code 0, we are good
+    if [[ $LOCAL_SUCCESS == 0 ]]; then
+        echo "=== PASSED TEST: ${TEST} ==="
+    # if a file does not run any tests, pytest reports exit code of 5
+    # since we skip full modules in our skipping strategy, this is common
+    elif [[ $LOCAL_SUCCESS == 5 ]]; then,
```
Comment (andy-neuma) [inline .github/scripts/run-tests:137]: yeah ... maybe we just echo "=== SKIPPED TEST: ${TEST} ===" ?

**Review by andy-neuma — APPROVED** [review]
thanks

File: .github/scripts/run-tests (hunk 3 above)
Comment (robertgshaw2-neuralmagic) [reply .github/scripts/run-tests]: updated

**Pull request merged**